src/java.base/share/classes/java/util/stream/Gatherers.java

```diff
@@ -0,0 +1,696 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package java.util.stream;
+
+import jdk.internal.access.SharedSecrets;
+import jdk.internal.javac.PreviewFeature;
+import jdk.internal.vm.annotation.ForceInline;
+
+import java.util.ArrayDeque;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.Semaphore;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Gatherer.Integrator;
+import java.util.stream.Gatherer.Downstream;
+
+/**
+ * Implementations of {@link Gatherer} that implement various useful intermediate
+ * operations, such as windowing functions, folding functions,
+ * transforming elements concurrently, etc.
+ *
+ * @since 22
+*/
+@PreviewFeature(feature = PreviewFeature.Feature.STREAM_GATHERERS)
+public final class Gatherers {
+    private Gatherers() { } // This class is not intended to be instantiated
+
+    // Public built-in Gatherers and factory methods for them
+
+    /**
+     * Returns a Gatherer that gathers elements into windows
+     * -- encounter-ordered groups of elements -- of a fixed size.
+     * If the stream is empty then no window will be produced.
+     * The last window may contain fewer elements than the supplied window size.
+     *
+     * <p>Example:
+     * {@snippet lang = java:
+     * // will contain: [[1, 2, 3], [4, 5, 6], [7, 8]]
+     * List<List<Integer>> windows =
+     *     Stream.of(1,2,3,4,5,6,7,8).gather(Gatherers.windowFixed(3)).toList();
+     * }
+     *
+     * @implSpec Each window produced is an unmodifiable List; calls to any
+     * mutator method will always cause {@code UnsupportedOperationException}
+     * to be thrown. There are no guarantees on the implementation type or
+     * serializability of the produced Lists.
+     *
+     * @apiNote For efficiency reasons, windows may be allocated contiguously
+     *          and eagerly. This means that choosing large window sizes for
+     *          small streams may use excessive memory for the duration of
+     *          evaluation of this operation.
```

Comment:
```suggestion
     *          small streams may use excessive memory during
     *          evaluation of this operation.
```
(Same for other methods.)